### PR TITLE
Migrated examples to OpenVINO 2023.0 and added tests for them

### PR DESCRIPTION
**Pull request by alexsu52**
### Changes
- Updated .gitignore to exude example's artifacts
- Migrated examples to OpenVINO 2023.0.
- Bump up ultralytics to 8.0.43 for yolo8 examples
- Fixed ssd300_vgg16 example
- Fixed yolo8 examples
- Added tests/cross_fw/examples/test_examples.py
- Updated Makefile

### Reason for changes
- Migrated examples to the latest OpenVINO.
- Regular example validation.

### Related tickets
N/A

### Tests
tests/cross_fw/examples/test_examples.py

Run tests via make: `make test-examples`

### CODE REVIEW

**Review by nikita-savelyevv — CHANGES_REQUESTED** [review]

File: examples/post_training_quantization/torch/ssd300_vgg16/main.py [hunk 1]
```diff
@@ -14,6 +14,10 @@
 import subprocess
 from pathlib import Path
 
+# NNCF should be imported before import torch
```
Comment (KodiaqQ) [inline examples/post_training_quantization/torch/ssd300_vgg16/main.py:17]: Why it should be?

File: tests/cross_fw/examples/test_examples.py
Comment (KodiaqQ) [inline tests/cross_fw/examples/test_examples.py:1]: Which kind of test is it? Nightly? Weekly? Are there any successful runs on CI?

File: tests/cross_fw/examples/test_examples.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    example_scope = load_json(EXAMPLE_SCOPE_PATH)
+    for example_name, example_params in example_scope.items():
+        yield pytest.param(example_name, example_params, id=example_name)
+
+
+@pytest.mark.parametrize("example_name, example_params", example_test_cases())
+def test_examples(tmp_path, example_name, example_params, backends_list, is_check_performance):
+    backend = example_params["backend"]
+    skip_if_backend_not_selected(backend, backends_list)
+    venv_path = create_venv_with_nncf(tmp_path, "pip_e_local", "venv", set([backend]))
```
Comment (nikita-savelyevv) [inline tests/cross_fw/examples/test_examples.py:47]: Could we somehow avoid creating an environment every time the test case is run? For example after every test run, clean up the example-specific packages / restore the backed up environment.

If run on CI the current approach would take too much time.

File: examples/post_training_quantization/torch/ssd300_vgg16/main.py (hunk 1 above)
Comment (nikita-savelyevv) [reply examples/post_training_quantization/torch/ssd300_vgg16/main.py:17]: It should say

`# nncf.torch must be imported before torchvision`

Please see ticket 109653.

File: Makefile [hunk 3]
```diff
@@ -42,15 +47,15 @@ install-openvino-test:
...
-	pip install -r examples/experimental/openvino/bert/requirements.txt
-	pip install -r examples/experimental/openvino/yolo_v5/requirements.txt
 	pip install git+https://github.com/openvinotoolkit/open_model_zoo.git@dcbf53280a95dae3c6538689bafe760470f08ec2#subdirectory=tools/model_tools
 
 install-openvino-dev: install-openvino-test install-pre-commit install-pylint
-	pip install -r examples/post_training_quantization/openvino/mobilenet_v2/requirements.txt
-	pip install -r examples/post_training_quantization/openvino/quantize_with_accuracy_control/requirements.txt
-	pip install -r examples/post_training_quantization/openvino/yolov8/requirements.txt
-	pip install -r examples/post_training_quantization/openvino/yolov8_quantize_with_accuracy_control/requirements.txt
+	pip install -r examples/experimental/openvino/bert/requirements.txt
```
[the remark this answers is not in the document]
Comment (alexsu52) [reply Makefile:53]: Thanks for the comment! I absolutely agree that this is potential point of getting a broken environment. In this PR I removed example's requiremnets from install-`backend`-test to avoid that case in a test environment. In future we should think about how to rework makefile to avoid such case in the dev environment as well but support pylint check in the same time.

File: .gitignore [hunk 4]
```diff
@@ -117,13 +117,21 @@ nncf_debug/
...
+examples/post_training_quantization/openvino/quantize/mobilenet_v2/mobilenet_v2_*
 examples/post_training_quantization/tensorflow/mobilenet_v2/mobilenet_v2_*
 examples/post_training_quantization/torch/mobilenet_v2/mobilenet_v2_*
+examples/post_training_quantization/torch/ssd300_vgg16/ssd300_vgg16_*
+examples/post_training_quantization/openvino/quantize_with_accuracy_control/anomaly_stfpm/stfpm_*
+examples/post_training_quantization/openvino/quantize/yolov8/yolov8n*
+examples/post_training_quantization/openvino/quantize_with_accuracy_control/yolov8/yolov8n*
 examples/**/runs/**
 examples/**/results/**
+compressed_graph.dot
```
[the remark this answers is not in the document]
Comment (alexsu52) [reply .gitignore:129]: Thanks for the comment! I absolutely agree again with your comment. I opened the PR https://github.com/openvinotoolkit/nncf/pull/1408 to do this, but it did not pass the review. This solution is workaround.

File: examples/post_training_quantization/torch/ssd300_vgg16/main.py (hunk 1 above)
Comment (alexsu52) [reply examples/post_training_quantization/torch/ssd300_vgg16/main.py:17]: Done

File: tests/cross_fw/examples/test_examples.py
Comment (alexsu52) [reply tests/cross_fw/examples/test_examples.py:1]: I suggested to add example tests to the nightly scope. Ticket 113243

File: examples/post_training_quantization/torch/ssd300_vgg16/main.py [hunk 5]
```diff
@@ -228,9 +152,15 @@ def main():
     quantized_model = nncf.quantize(model, calibration_dataset)
 
     # Convert to OpenVINO
-    input_shape = [3, 480, 480]
-    ov_model = mo.convert_model(model.cpu(), input_shape=[-1] + input_shape)
-    ov_quantized_model = mo.convert_model(quantized_model.cpu(), input_shape=[-1] + input_shape)
+    dummy_input = torch.randn(1, 3, 480, 480)
+
+    fp32_onnx_path = f"{ROOT}/ssd300_vgg16_fp32.onnx"
+    torch.onnx.export(model.cpu(), dummy_input, fp32_onnx_path)
```
[the remark this answers is not in the document]
Comment (alexsu52) [reply examples/post_training_quantization/torch/ssd300_vgg16/main.py:158]: `ov.convert_model` does not support torch quantized models in OpenVINO 2023.0. Thus I had to add the additional step of converting the quantized model to onnx. I did it for both original and quantized model for consistency.

File: tests/cross_fw/examples/test_examples.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    env = os.environ.copy()
+    env["PYTHONPATH"] = str(PROJECT_ROOT)  # need this to be able to import from tests.* in run_example.py
+
+    metrics_file_path = tmp_path / "metrics.json"
+    python_executable_with_venv = get_python_executable_with_venv(venv_path)
+    run_example_py = EXAMPLE_TEST_ROOT / "run_example.py"
+    run_cmd_line = f"{python_executable_with_venv} {run_example_py} --name {example_name} --output {metrics_file_path}"
+    subprocess.run(run_cmd_line, check=True, shell=True, env=env, cwd=PROJECT_ROOT)
+
+    mesured_metrics = load_json(metrics_file_path)
```
[the remark this answers is not in the document]
Comment (alexsu52) [reply tests/cross_fw/examples/test_examples.py:63]: done

File: examples/post_training_quantization/openvino/quantize/yolov8/requirements.txt [hunk 7]
```diff
@@ -0,0 +1,3 @@
+ultralytics==8.0.43
+onnx
```
[the remark this answers is not in the document]
Comment (alexsu52) [reply examples/post_training_quantization/openvino/quantize/yolov8/requirements.txt:2]: removed

File: tests/cross_fw/examples/test_examples.py (hunk 2 above)
Comment (alexsu52) [reply tests/cross_fw/examples/test_examples.py:47]: Do you have an estimate of how much faster it will be?

File: tests/cross_fw/examples/example_scope.json [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+        }
+    },
+    "post_training_quantization_torch_ssd300_vgg16": {
+        "backend": "torch",
+        "requirements": "examples/post_training_quantization/torch/ssd300_vgg16/requirements.txt",
+        "cpu": "Intel(R) Core(TM) i9-10980XE CPU @ 3.00GHz",
+        "accuracy_metrics": {
+            "fp32_mAP": 0.11208358407020569,
+            "int8_mAP": 0.11117147654294968,
+            "accuracy_drop": 0.000912107527256012
```
[the remark this answers is not in the document]
Comment (alexsu52) [reply tests/cross_fw/examples/example_scope.json]: done

**Review by KodiaqQ — APPROVED** [review]

File: examples/post_training_quantization/torch/ssd300_vgg16/main.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (nikita-savelyevv) [reply examples/post_training_quantization/torch/ssd300_vgg16/main.py:158]: Thanks for the explanation

File: examples/post_training_quantization/openvino/yolov8/requirements.txt [hunk 9]
```diff
@@ -0,0 +1,2 @@
+ultralytics==8.0.43
+openvino-dev==2023.0.0
```
Comment (nikita-savelyevv) [inline examples/post_training_quantization/openvino/yolov8/requirements.txt:2]: After removal of `onnx` as a requirement in the last commit, this test now fails for me with 

`ONNX: export failure ❌ 2.0s: No module named 'onnx'`

error. Could you please check again?

File: examples/post_training_quantization/openvino/yolov8_quantize_with_accuracy_control/requirements.txt [hunk 10]
```diff
@@ -0,0 +1,2 @@
+ultralytics==8.0.43
+openvino-dev==2023.0.0
```
Comment (nikita-savelyevv) [inline examples/post_training_quantization/openvino/yolov8_quantize_with_accuracy_control/requirements.txt:2]: Same issue with `onnx`

File: tests/cross_fw/examples/example_scope.json [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+            "int8_model_size": 2.69466495513916,
+            "model_compression_rate": 3.1900953474371994
+        }
+    },
+    "post_training_quantization_torch_mobilenet_v2": {
+        "backend": "torch",
+        "requirements": "examples/post_training_quantization/torch/mobilenet_v2/requirements.txt",
+        "cpu": "Intel(R) Core(TM) i9-10980XE CPU @ 3.00GHz",
+        "accuracy_metrics": {
+            "fp32_top1": 0.8761783439490446,
```
Comment (nikita-savelyevv) [inline tests/cross_fw/examples/example_scope.json:112]: I got a different resulting metric when running this test:
```
>           assert measured_metrics[name] == pytest.approx(value, abs=ACCURACY_TOLERANCE)
E           assert 0.9864968152866243 == 0.8761783439490446 ± 2.0e-03
E             comparison failed
E             Obtained: 0.9864968152866243
E             Expected: 0.8761783439490446 ± 2.0e-03
```

Could you please check again?

Comment (alexsu52) [reply tests/cross_fw/examples/example_scope.json:112]: Surprised, Could you share how you run test? I used `make test-examples`.
[log.txt](https://github.com/openvinotoolkit/nncf/files/11757668/log.txt)

File: examples/post_training_quantization/openvino/yolov8/requirements.txt (hunk 9 above)
Comment (alexsu52) [reply examples/post_training_quantization/openvino/yolov8/requirements.txt:3]: done

File: tests/cross_fw/examples/example_scope.json (hunk 11 above)
Comment (nikita-savelyevv) [reply tests/cross_fw/examples/example_scope.json:112]: I run it via `PYTHONPATH=../../.. pytest . -s`
But I get the same failure when running according to your command: [test_examples_log.txt](https://github.com/openvinotoolkit/nncf/files/11760272/test_examples_log.txt)

I see your run takes almost two hours, while mine is only half an hour. Do you run on a CPU?

Comment (alexsu52) [reply tests/cross_fw/examples/example_scope.json]: Thanks, updated.

**Review by nikita-savelyevv — APPROVED** [review]

**Comment by alexsu52** [comment]
run pytorch pre-commit tests
